File: www/main/cgi-bin/delete_resource.php
#!/usr/bin/env php
<?php
// En-têtes HTTP requis
header('Content-Type: text/plain; charset=utf-8');

$upload_dir = './uploads/';

// Vérifier la méthode de requête
if ($_SERVER['REQUEST_METHOD'] === 'DELETE') {
    // Parse la requête pour récupérer le paramètre 'resource'
    parse_str(file_get_contents("php://input"), $_DELETE);
    $resource = isset($_GET['resource']) ? $_GET['resource'] : '';

    $file_path = $uploads_dir . $resource;

    // Vérifier que la ressource est valide
    if (!empty($resource) && file_exists($file_path)) {
        if (unlink($file_path)) {
            echo "Resource '$resource' has been deleted successfully.";
        } else {
            echo "Failed to delete the resource '$resource'.";
        }
    } else {
        echo "Resource '$resource' does not exist.";
    }
} else {
    echo "Invalid request method. Use DELETE.";
}
?>

File: www/main/cgi-bin/delete_resource.py
#!/usr/bin/env python3

import os
import urllib.parse

# En-têtes HTTP requis
print("Content-Type: text/plain; charset=utf-8\r\n\r\n")

upload_dir = './www/main/cgi-bin/uploads/'

# Vérifier la méthode de requête
if os.environ['REQUEST_METHOD'] == 'DELETE':
    # Extraire la chaîne de requête de l'URL
    query_string = os.environ.get('QUERY_STRING', '')
    params = urllib.parse.parse_qs(query_string)
    resource = params.get('resource', [''])[0]  # Récupère la première valeur de 'resource'

    file_path = os.path.join(upload_dir, resource)
    #print(f"file_path: {file_path}")
    #print(f"resource: {resource}")

    # Vérifier que la ressource est valide
    if resource and os.path.exists(file_path):
        try:
            os.remove(file_path)
            print(f"Resource '{resource}' has been deleted successfully.")
        except Exception as e:
            print(f"Failed to delete the resource '{resource}': {e}")
    else:
        print(f"Resource '{resource}' does not exist.")
else:
    print("Invalid request method. Use DELETE.")
